zipkin-storage/mysql/src/main/java/zipkin/storage/mysql/MySQLSpanStore.java

```diff
@@ -271,7 +271,8 @@ private Endpoint endpoint(Record a) {
           .selectDistinct(ZIPKIN_SPANS.NAME)
           .from(ZIPKIN_SPANS)
           .join(ZIPKIN_ANNOTATIONS)
-          .on(ZIPKIN_SPANS.TRACE_ID.eq(ZIPKIN_ANNOTATIONS.TRACE_ID))
+          .on(ZIPKIN_SPANS.TRACE_ID_HIGH.eq(ZIPKIN_ANNOTATIONS.TRACE_ID_HIGH))
```

Comment:
guess I'm wondering if this needs to be refactored to use Schema.joinCondition() or similar?

Reply:
I think the index is not available without trace_id_high when seach or join because the trace_id_high is the first key of the INDEX(`trace_id_high`, `trace_id`, `id`)  and INDEX(`trace_id_high`, `trace_id`). So the trace_id_high added can optimize the join or the query. I think it is necessary to refactor the Schema.joinCondition(). Or change the index?